python/plot_multiple_index.py

```diff
@@ -0,0 +1,269 @@
+#names = ['Global', '90N-50N', '50N-20N', '20N-20S', '20S-50S', '50S-90S']
+
+print 'salil', regs
+print 'salil', names
+
+colors = ['b', 'g', 'r', 'c', 'm', 'y']
+
+x = mpl.get_backend()
+print 'backend: ', x
+ 
+def plot_multiple_index   (indir,
+			   casename,
+			   field_name,
+			   interp_grid,
+			   interp_method,
+			   begin_yr,
+			   end_yr,
+			   begin_month,
+			   end_month,
+			   regs,
+			   names,
+			   index_set_name,
+			   aggregate,
+			   no_ann,
+			   stdize,
+			   debug = False):
+
+	n_reg = len(regs)
+
+	for i,reg in enumerate(regs):
+		print __name__, 'casename: ', casename
+		area_seasonal_avg, n_months_season, units = get_reg_seasonal_avg (
+								  indir 	= indir,
+								  casename 	= casename, 
+								  field_name 	= field_name,
+								  interp_grid 	= interp_grid,
+								  interp_method = interp_method,
+								  begin_yr 	= begin_yr,
+								  end_yr 	= end_yr,
+								  begin_month 	= begin_month,
+								  end_month 	= end_month,
+								  reg 		= reg,
+								  aggregate 	= aggregate,
+								  debug 	= debug)
+
+		if i == 0: plot_ts = numpy.zeros((n_reg, area_seasonal_avg.shape[0]))
+
+		plot_ts[i, :] = area_seasonal_avg 
+
+		if aggregate == 0 and no_ann == 1:
+			area_seasonal_avg_no_ann = remove_seasonal_cycle_monthly_data(plot_ts[i, :], n_months_season, debug = debug)
+			plot_ts[i, :] = area_seasonal_avg_no_ann
+		
+		if stdize == 1:
+			area_seasonal_avg_stddize = standardize_time_series(plot_ts[i, :])
+			plot_ts[i, :] = area_seasonal_avg_stddize
+
+
+        if debug: print __name__, 'plot_ts: ', plot_ts
+
+	plot_ts_mean = numpy.mean(plot_ts, axis = 1)
+	plot_ts_stddev = numpy.std(plot_ts, axis = 1)
+
+	if debug: print __name__, 'plot_ts_mean: ', plot_ts_mean
+
+	f, ax = plt.subplots(n_reg, sharex = True, figsize=(8.5,11))
+
+	nt = plot_ts.shape[1]
+
+
+	f.text(0.5, 0.04, 'Model Year', ha='center', fontsize = 24)
+
+	f.text(0.04, 0.5, field_name + ' (' + units + ')', va='center', rotation='vertical', fontsize = 16)
+
+	season = get_season_name(begin_month, end_month)
+	plt.suptitle('Monthly ' + index_set_name + ' index', fontsize = 24)
+
+        if aggregate == 1: 
+		plot_time = numpy.arange(0,nt) + begin_yr
+	else:
+		plot_time = numpy.arange(0,nt)
+
+	if debug: print __name__, 'plot_time: ', plot_time
+
+	for i,name in enumerate(names):
+		min_plot = numpy.amin(plot_ts[i, :])
+		max_plot = numpy.amax(plot_ts[i, :])
+
+		y_axis_ll = min_plot - 0.5*numpy.std(plot_ts[i, :])
+		y_axis_ul = max_plot + 0.5 * numpy.std(plot_ts[i,:])
+
+		ax[i].axis([plot_time[0],plot_time[-1], y_axis_ll, y_axis_ul])
+
+		print 'plot_time[0],plot_time[-1], 1.1*min_plot, 1.1*max_plot: ', \
+			plot_time[0],plot_time[-1], 1.1*min_plot, 1.1*max_plot
+
+		mean_line_plot = numpy.zeros(nt) + plot_ts_mean[i]
+		mean_line, = ax[i].plot(plot_time, mean_line_plot, color = 'black', linewidth = 1.0, label = 'Mean')
+
+		if begin_month == 0 and end_month == 11 and aggregate == 0:
+			bw   = 13
```

Comment:
@salilmahajan could you explain why a 1-yr running mean was chosen for the NINO3,4,3.4 indices?  Can this be changed to 5-months, which is the NINO index algorithm (http://www.cgd.ucar.edu/staff/trenbert/trenberth.papers/defnBAMS.pdf)